front/src/common/Map/WarpedMap/WarpedMap.tsx

```diff
@@ -0,0 +1,200 @@
+import OrderedLayer, { OrderedLayerProps } from 'common/Map/Layers/OrderedLayer';
+import { genLayerProps } from 'common/Map/Layers/OSM';
+import osmBlankStyle from 'common/Map/Layers/osmBlankStyle';
+import { Viewport } from '../../../reducers/map';
+import { AllowancesSettings, Train } from '../../../reducers/osrdsimulation/types';
+
+const OSRD_LAYER_ORDERS: Record<LayerType, number> = {
+  buffer_stops: LAYER_GROUPS_ORDER[LAYERS.BUFFER_STOPS.GROUP],
+  detectors: LAYER_GROUPS_ORDER[LAYERS.DETECTORS.GROUP],
+  signals: LAYER_GROUPS_ORDER[LAYERS.SIGNALS.GROUP],
+  switches: LAYER_GROUPS_ORDER[LAYERS.SWITCHES.GROUP],
+  track_sections: LAYER_GROUPS_ORDER[LAYERS.TRACKS_SCHEMATIC.GROUP],
+  // Unused:
+  catenaries: 0,
+  lpv: 0,
+  lpv_panels: 0,
+  routes: 0,
+  speed_sections: 0,
+  errors: 0,
+};
+
+/**
+ * This component handles displaying warped data. The data must be warped before being given to this component.
+ * Check `SimulationWarpedMap` to see an example use case.
+ */
+const WarpedMap: FC<{
+  bbox: BBox2d;
+  osrdLayers: Set<LayerType>;
+  boundingBox?: LngLatBoundsLike;
+  // Data to display on the map (must be transformed already):
+  osrdData: Partial<Record<LayerType, FeatureCollection>>;
+  osmData: Record<string, FeatureCollection>;
+  trainsPositions?: (TrainPosition & { train: Train; isSelected?: boolean })[];
+  itinerary?: Feature<LineString>;
+  allowancesSettings?: AllowancesSettings;
+}> = ({
+  bbox,
+  osrdLayers,
+  osrdData,
+  osmData,
+  trainsPositions,
+  itinerary,
+  boundingBox,
+  allowancesSettings,
+}) => {
+  const prefix = 'warped/';
+  const [mapRef, setMapRef] = useState<MapRef | null>(null);
+  const { mapStyle, layersSettings, showIGNBDORTHO } = useSelector((s: RootState) => s.map);
+  const [viewport, setViewport] = useState<Viewport | null>(null);
+
+  // Main OSM and OSRD data:
+  const layerContext: LayerContext = useMemo(
+    () => ({
+      colors: colors[mapStyle],
+      signalsList: ALL_SIGNAL_LAYERS,
+      symbolsList: ALL_SIGNAL_LAYERS,
+      sourceLayer: 'geo',
+      prefix: '',
+      isEmphasized: false,
+      showIGNBDORTHO,
+      layersSettings,
+    }),
+    [colors, mapStyle, showIGNBDORTHO, layersSettings]
+  );
+  const osrdSources = useMemo(
+    () =>
+      Array.from(osrdLayers).map((layer) => ({
+        source: layer,
+        order: OSRD_LAYER_ORDERS[layer],
+        id: `${prefix}geo/${layer}`,
+        layers: SourcesDefinitionsIndex[layer](layerContext, prefix).map(
+          (props) => omit(props, 'source-layer') as typeof props
+        ),
+      })),
+    [osrdLayers]
+  );
+  const osmSources = useMemo(
+    () =>
+      groupBy(
+        (
+          genLayerProps(
+            mapStyle,
+            LAYER_GROUPS_ORDER[LAYERS.BACKGROUND.GROUP]
+          ) as (OrderedLayerProps & {
+            'source-layer': string;
+          })[]
+        ).filter((layer) => !layer.id?.match(/-en$/)),
```

Comment:
What do you do here ?
```suggestion
        ).filter((layer) => !layer.id?.match(/-en$/)),
```

Reply:
There are various OSM layers doubled with a `-en` suffixed clone. You can see this one for instance:

https://github.com/osrd-project/osrd/blob/7c7f180b11b475589d248a2df45fd44b4d09ed9a/front/src/assets/mapstyles/OSMStyle.json#L1457

I'll add a comment to clarify this.

Reply:
Thanks